src/xcode/ENA/ENA/Source/Services/CCLService/DCCUIText.swift

```diff
@@ -0,0 +1,273 @@
+
+	static let dateFormatter: ISO8601DateFormatter = .iso8601DateFormatter()
+	static let localDateFormatter: DateFormatter = .localDateFormatter()
+	static let localDateTimeFormatter: DateFormatter = .localDateTimeFormatter()
+	static let outputDateFormatter: DateFormatter = .outputDateFormatter()
+	static let outputDateTimeFormatter: DateFormatter = .outputDateTimeFormatter()
+	
+	// MARK: - Internal
+
+	func localized(languageCode: String? = Locale.current.languageCode) -> String? {
+		switch type {
+		case UITextType.string:
+			return localizedSingleFormatText(languageCode: languageCode)
+		case UITextType.plural:
+			return localizedPluralFormatText(languageCode: languageCode)
+		case UITextType.systemTimeDependent:
+			return localizedSystemTimeDependentFormatText(languageCode: languageCode)
+		default:
+			return nil
+		}
+	}
+	
+	// MARK: - Private
+	
+	private func localizedSingleFormatText(languageCode: String?) -> String? {
+		var formatText = ""
+		
+		// use language code, if there is no property for the language code, en shall be used
+		if let localizedFormatText = localizedText?[languageCode ?? "en"]?.value as? String {
+			formatText = localizedFormatText
+		} else if let fallbackLocalizedFormatText = localizedText?["de"]?.value as? String { // if en is not available, de shall be used
+			formatText = fallbackLocalizedFormatText
+		} else {
+			return nil
+		}
+		
+		// replacing %s with %@, %1$s with %1$@ and so on
+		formatText = formatText.replacingOccurrences(of: "%(\\d\\$)?s", with: "%$1@", options: .regularExpression, range: nil)
+		
+		if let parameters = parameters.value as? [[String: Any]] {
+			if parameters.isEmpty {
+				// regular text without placeholders
+				return formatText
+			} else { // regular text with placeholder
+				var mappedParameters: [DCCUITextParameter] = []
+				
+				// we could get multiple parameters
+				for parameter in parameters {
+					guard let type = parameter["type"] as? String, let value = parameter["value"] else {
+						return nil
+					}
+					mappedParameters.append(DCCUITextParameter(type: type, value: value))
+				}
+				// text shall be determined by passing formatText and formatParameters to a printf-compatible format function
+				return formattedTextWithParameters(formatText: formatText, parameters: mappedParameters)
+			}
+		} else {
+			// regular text without placeholders
+			return formatText
+		}
+	}
+	
+	private func localizedPluralFormatText(languageCode: String?) -> String? {
+		guard let formatText = localizedFormatText(languageCode: languageCode) else {
+			return nil
+		}
+		
+		if let parameters = parameters.value as? [[String: Any]] {
+			if parameters.isEmpty {
+				// text without parameters
+				if let textDescriptorQuantity = quantity {
+					return quantityBasedFormatText(formatText: formatText, quantity: textDescriptorQuantity)
+				}
+			} else {
+				var mappedParameters: [DCCUITextParameter] = []
+				
+				// we could get multiple parameters
```

Comment:
This part is the same like in line 84. Could be extracted to make it better readable and maintainable.